File: benchmarking/tests/report_repair_01/solution/solution.py
import math
import itertools
from functools import reduce

def repair(expenses):

    """
    >>> repair([1721, 979, 979, 366, 299, 675, 1456])
    514579
    """

    for expenses in itertools.combinations(expenses, 2):
        if sum(expenses) == 2020:
            # return math.prod(expenses) # Python 3.8
            return reduce(lambda x, y: x * y, expenses)

if __name__ == '__main__':
    import doctest
    doctest.testmod()

    # data = [int(line) for line in open('adventofcode.input.txt')]
    # print(repair(data))

File: benchmarking/tests/report_repair_01/solution/solution.js
function repair(expenses) {

    for (let index1 = 0; index1 < expenses.length - 1; index1 += 1) {
        let expenses1 = expenses[index1];
        for (let index2 = index1 + 1; index2 < expenses.length; index2 += 1) {
            let expenses2 = expenses[index2];
            if (expenses1 + expenses2 === 2020) {
                return expenses1 * expenses2;
            }
        }
    }

}

// console.log(repair([1721, 979, 979, 366, 299, 675, 1456]));
// 514579
